plugin/backend.go

```diff
@@ -94,6 +100,35 @@ func (b *GcpAuthBackend) IAMClient(s logical.Storage) (*iam.Service, error) {
 	return client.(*iam.Service), nil
 }
 
+// IAMCredentialsClient returns a new IAM Service Account Credentials client.
+// This client talks to the IAM Service Credentials endpoint, for signing JWTs.
+//
+// See:
+// https://pkg.go.dev/google.golang.org/api@v0.45.0/iamcredentials/v1#pkg-overview
+//
+// The client is cached.
+func (b *GcpAuthBackend) IAMCredentialsClient(s logical.Storage) (*iamcredentials.Service, error) {
```

Comment:
Should we change the call in [here](https://github.com/hashicorp/vault-plugin-auth-gcp/blob/master/plugin/path_login.go#L291) to use `IAMCredentialsClient` instead, and is there a need to keep `IAMClient` around?

Reply:
We discussed this internally, but for posterity: 

- `IAMClient` and `IAMCredentialsClient` return different clients for different APIs. They are both valid to include here, and only the `SignJWT` method from `IAMClient` is being deprecated, while the other methods in use remain supported.